util/src/org/commcare/util/cli/EntityListSubscreen.java

```diff
@@ -0,0 +1,137 @@
+package org.commcare.util.cli;
+
+import org.commcare.suite.model.Detail;
+import org.commcare.suite.model.DetailField;
+import org.commcare.suite.model.SessionDatum;
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.javarosa.core.model.instance.TreeReference;
+
+import java.io.PrintStream;
+import java.util.Vector;
+
+/**
+ * The entity list subscreen handles actually displaying the list of dynamic entities to the
+ * user to be displayed and chosen during a <datum> selection
+ *
+ * Created by ctsims on 8/20/2015.
+ */
+public class EntityListSubscreen extends Subscreen<EntityScreen> {
+
+    EvaluationContext mSessionContext;
+
+    private final int SCREEN_WIDTH = 100;
+
+    private TreeReference[] mChoices;
+    private String[] rows;
+    private String mHeader;
+
+
+    public EntityListSubscreen(Detail shortDetail, SessionDatum needed, EvaluationContext context) throws CommCareSessionException{
+        mHeader = this.createHeader(shortDetail, context);
+
+        Vector<TreeReference> references = inflateReference(needed.getNodeset(), context);
+
+        rows = new String[references.size()];
+
+        int i = 0;
+        for (TreeReference entity : references) {
+            rows[i] = createRow(entity, shortDetail, context);
+            ++i;
+        }
+
+
+        this.mChoices = new TreeReference[references.size()];
+        references.copyInto(mChoices);
+    }
+
+    private String createRow(TreeReference entity, Detail shortDetail, EvaluationContext ec) {
+        EvaluationContext context = new EvaluationContext(ec, entity);
+
+        DetailField[] fields = shortDetail.getFields();
+
+        StringBuilder row = new StringBuilder();
+        int i = 0;
+        for (DetailField field : fields) {
+            Object o = field.getTemplate().evaluate(context);
+            String s;
+            if (!(o instanceof String)) {
+                s = "";
+            } else {
+                s = (String)o;
+            }
+
+            int widthHint = SCREEN_WIDTH / fields.length;
```

Comment:
Non-blocking (and I realize you're just moving this, I just haven't looked at much of the CLI code yet) but I'm surprised you're bothering with template width hints at all, doesn't seem like they'd ever translate well to an ASCII interface.